Vaccination Certificate (feature main branch)

d4rken opened the pull request:
WIP :tea:

d4rken committed:
Vaccination skeleton classes. (#3010)

chiljamgossow commented on Corona-Warn-App/src/main/java/de/rki/coronawarnapp/vaccination/core/VaccinatedPerson.kt:
```diff
@@ -0,0 +1,32 @@
+data class VaccinatedPerson(
+    val vaccinationCertificates: Set<VaccinationCertificate>,
+    val proofCertificates: Set<ProofCertificate>,
+    val isRefreshing: Boolean,
```
What does refreshing mean in this context?

d4rken replied:
That this data is currently being updated, e.g. polling servers for a new ProofCertificate.
Use case:
User presses refresh button in UI, UI wants to display a progress spinner.
This can't be only handled on viewmodel level, as there will also be automatic refreshes/updates via workers, that could overlap with the user having the UI open.

chiljamgossow replied:
Can we call it isUpdatingData or the like to make more clear? My first association was refreshing the vaccination.

d4rken replied:
Yes, will do in another PR.

chiljamgossow commented on Corona-Warn-App/src/main/java/de/rki/coronawarnapp/vaccination/core/VaccinationCertificate.kt:
```diff
@@ -0,0 +1,27 @@
+    val vaccinatedAt: Instant
+
+    val vaccinationName: String
+    val vaccinationManufacturer: String
```
vaccineName and vaccineManufacturer

d4rken committed:
Merge branch 'release/2.3.x' into feature/vaccination-main

d4rken committed:
Improve VaccinationCertificate property names.

chiljamgossow commented:
Looks well structured overall, thanks for the preparation

mtwalli committed:
Vaccination  Details (EXPOSUREAPP-6721) (#3015)

* Setup screen

* Screen ui and wiring

* Update VaccinationUIModule.kt

* Create new strings file

* Bind view with logic

* Apply parallax effect

* Fix lint error

* Handle complete and incomplete vaccination

* Add to actual graph

* PR comments

* Update test_nav_graph.xml

* Fix background

* Rename to vaccinationCertificateId

* adapt changes

* lint

* Update VaccinationRepository.kt

* Deletion dialog

* Adjust strings

d4rken committed:
Merge branch 'release/2.3.x' into feature/vaccination-main

jurajkusnier committed:
Home Screen - Registration card (EXPOSUREAPP-6724) (#3061)

* Vaccination Certificate Register Home Card

* Lint format

* Update CreateVaccinationHomeCard

* Update card layout

d4rken committed:
Vaccination Repository Part 1 (EXPOSUREAPP-6729) (#3058)

* Repository WIP

* Add vaccination location

* Adjust label and type for country.

* Vaccination storage WIP, Draft 2

* Vaccination storage WIP, Draft 5

* Vaccination storage WIP, Draft 6

* Revert gradle change.

* Vaccination storage WIP, Draft 7

* Adjust vaccination certificate json schema.

* Extend proof certificate with known attributes, adjust unit tests.

* Error handling, TODOs, more structure, storage and tests.

* LINTs

* Restructure storage approach, reparsing instead of duplicate data.

* Fix temporary issues.

d4rken committed:
Merge branch 'release/2.3.x' into feature/vaccination-main

# Conflicts:
#	Corona-Warn-App/src/main/res/values/strings.xml

d4rken committed:
Merge branch 'release/2.3.x' into feature/vaccination-main

d4rken committed:
Re-add strings that got lost during merge. (#3094)

mtwalli committed:
Vaccination details screenshots (EXPOSUREAPP-6738) (#3048)

* Vaccination details screenshots

* lint

* Update test

* Update sceenshots mock data

* Fix format

* Update ScreenShotter.kt

* Remove string

mtwalli committed:
Merge branch 'release/2.3.x' into dev/2.3-to-vaccination-main

d4rken committed:
Merge pull request #3109 from corona-warn-app/dev/2.3-to-vaccination-main

2.3 to vaccination main (DEV)

d4rken committed:
Merge branch 'release/2.3.x' into feature/vaccination-main

BMItr committed:
Vaccination Server - Retrofit interfaces & Module  (EXPOSUREAPP-6854, EXPOSUREAPP-6856, EXPOSUREAPP-7129) (#3099)

* Create some classes

* Get vaccination urls from env json

* Implemented api for obtaining value sets

* improved package structure, merge adjustments

* Adjusted path

* Share cache so one can clean it

* Added certificate retrofit

* Correction

* satisfy CI

* clean import

* fix test

* Fix test #2

* Dont use retrofit cache for more control

* Revert "Dont use retrofit cache for more control"

This reverts commit 44fed1f6d5258c00f1a5db72a329cfb91b6c60fa.

* Renamed vaccination url, Adjusted test

* Update prod json

* Changed name

* Updated debug menu

* Added new optional api

* Update DebugOptionsFragmentViewModelTest.kt

Co-authored-by: harambasicluka <64483219+harambasicluka@users.noreply.github.com>

chiljamgossow committed:
Vaccination qr code extraction (EXPOSUREAPP-6726) (#3107)

* extractor

* decoding

* decoding, error handling

* clean up

* klint

* detekt

* simplify condition

* comment failing unit test

* unit test

* change lib

* Unit tests for RA/PCRCoronaTest.isFinal

* Add license text.

* add header

* clean up

* merge

* comments

* klint

* revert version change

Co-authored-by: Matthias Urhahn <matthias.urhahn@sap.com>

LukasLechnerDev committed:
Screen: Vaccination List (EXPOSUREAPP-6734) (#3082)

* Create first version of incomplete vaccination list screen with mock data

* Add UiState for VaccinationListFragment.kt

* Set Toolbar Overlay

* Fine tune fragment_vaccination_list.xml

* Use ModularAdapter

* Add click listener to navigate to detail fragment

* Fix nav_graph issue

* Adjust layouts for night mode

* Fix vaccination card colors

* Fix mock data

* Revert changes in Project.xml

* Add TODO

* Show blue background on complete vaccination status

* Show subtitle on complete vaccination status

* Add VaccinationListCertificateCardItem.kt

* Show Refresh button on complete vaccination status

* Add fragment in vaccination_nav_graph.xml

* Connect Vaccination List ViewModel to Vaccination Repository

* Create new mock data

* Adapt VaccinationTestFragment.kt

* Try to make sonar happy

* Use VaccinationCertificate and ProofCertificate interfaces instead of VaccinationCertificateV1 and ProofCertificateV1

* Adjust title+subtitle alignment

* Adjust text size in vaccination_list_incomplete_top_card.xml

* Use isEligibleForProofCertificate boolean to show 'refresh' or 'register new vaccination' button

* Adjust subtitle color of vaccination_list_certificate_card.xml and vaccination_list_name_card.xml

* Fix package declaration of VaccinationListAdapter

* Update VaccinationListAdapter instead of re-creating a new one each time we observe a uiState change

* Refactor Navigation

* Show sample QR-code

* Add some layout improvements

* Adjust background height for COMPLETE and INCOMPLETE state

* Show both buttons in VaccinationListFragment - Refresh button and register Vaccination button

* Use separate flow to generate qrCode

* Refactor CardItems to be inlined into their ViewHolders

* Ignore onClick listener in equals() and hashCode() of VaccinationListVaccinationCardItem.kt to avoid re-drawings

* Fix Lint

* Fix Detekt

Co-authored-by: Mohamed <mohamed.metwalli@sap.com>
Co-authored-by: harambasicluka <64483219+harambasicluka@users.noreply.github.com>

d4rken committed:
Add home cards for vaccination status (EXPOSUREAPP-6730, EXPOSUREAPP-6731) (#3089)

* Add home cards for vaccination status.

* Fix instrumentation test.

* Fix instrumentation test, season 1, episode 2.

* fix icons

* Address PR comments

* Card label is not dynamic based on vaccine ID, it's a hardcoded name.

* Replace card graphics with new design.

Co-authored-by: Juraj Kusnier <jurajkusnier@gmail.com>
Co-authored-by: Juraj Kusnier <jurajkusnier@users.noreply.github.com>
Co-authored-by: harambasicluka <64483219+harambasicluka@users.noreply.github.com>

mtwalli committed:
Remove dummy data (#3138)

d4rken committed:
Merge branch 'release/2.3.x' into feature/vaccination-main

mtwalli committed:
Polishing (EXPOSUREAPP-6913) (#3135)

* Connect to actual graph

* Adapt new title

* Lint

* Connect to main screen cards

* Lint

* FixME :: 🛠️

* lint

* Keep testing logic

* Remove fix me

* Navigation

* Naming

chiljamgossow committed:
Vaccination QR code scan screen, error handling, navigation (EXPOSUREAPP-6726) (#3124)

* extractor

* decoding

* decoding, error handling

* clean up

* klint

* detekt

* simplify condition

* comment failing unit test

* unit test

* change lib

* Unit tests for RA/PCRCoronaTest.isFinal

* Add license text.

* add header

* clean up

* scan screen, error handling, navigation

* clean up

* klint detekt

* merge

* move strings

* comments

* klint

* revert version change

* fix fun name

* fix merge conflicts

* show dialog

* add navigation

* fix error dialog

* fix error dialog

* fix error dialog

* fix error dialog

* simplify

* merge main

* fix package

* review comments

* circular progress spinner

* fix test

Co-authored-by: Matthias Urhahn <matthias.urhahn@sap.com>

d4rken committed:
Automatic vaccination data updates (EXPOSUREAPP-6733) (#3119)

* Vaccination data updates, draft 1: Skeletons.

* Vaccination data updates, draft 2: Wiring calls to refresh and condition checks.

* Vaccination data updates, draft 3: Unit tests.

* Introduce unit test that checks application setup.

* Improve logging.

* LINTs

* Change mockk init.

* Add missing task factory registration.

Co-authored-by: Mohamed <mohamed.metwalli@sap.com>

d4rken committed:
Vaccination Repository Episode 2 (EXPOSUREAPP-6729) (#3139)

* Vaccination data updates, draft 1: Skeletons.

* Vaccination data updates, draft 2: Wiring calls to refresh and condition checks.

* Vaccination data updates, draft 3: Unit tests.

* Introduce unit test that checks application setup.

* Improve logging.

* LINTs

* Wrap the RawCOSEObject to prevent confusion, and help with conversion.
typealias does not prevent missuse on compiler level.
Kotlin value classes are still experimental :(.

* * Make the base45 a project utility class, align call pattern with base32 and base64. Additional unit tests.
* Align vaccination storage with COSE encoding/decoding. Unit tests.

* Change mockk init.

* Add missing task factory registration.

* Resolve merge conflicts.

Co-authored-by: Mohamed <mohamed.metwalli@sap.com>

d4rken committed:
Merge branch 'release/2.3.x' into feature/vaccination-main
